src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ApplicationService.kt

```diff
@@ -799,7 +802,13 @@ class ApplicationService(
       situation = submitApplication.situation?.toString()
       inmateInOutStatusOnSubmission = inmateDetails?.custodyStatus?.name
       this.apArea = apArea
-      this.cruManagementArea = apArea.defaultCruManagementArea
+      this.cruManagementArea = if (featureFlagService.getBooleanFlag("cas1-womens-estate-enabled") &&
+        submitApplication.isWomensApplication == true
+      ) {
+        cas1CruManagementAreaRepository.findByIdOrNull(Cas1CruManagementAreaEntity.WOMENS_ESTATE_ID) ?: apArea.defaultCruManagementArea
+      } else {
+        apArea.defaultCruManagementArea
+      }
```

Comment:
Should throw InternalServerErrorProblem if finding the women's estate CRU management area returns null:

`cas1CruManagementAreaRepository.findByIdOrNull(Cas1CruManagementAreaEntity.WOMENS_ESTATE_ID) ?: throw InternalServerErrorProblem("Could not find women's estate CRU Management Area Entity with ID ${Cas1CruManagementAreaEntity.WOMENS_ESTATE_ID}")`

Reply:
Done this.